packages/util/src/documentation-block.ts

```diff
@@ -0,0 +1,86 @@
+import { EOL } from 'os';
+import { zip } from './array';
+import { createLogger } from './logger';
+
+const logger = createLogger('documentation-block');
+/**
+ * @experimental This API is experimental and might change in newer versions. Use with caution.
+ * Transform strings and arguments to a string formatted as a documentation block.
+ * The formatting is block like so no leading or trailing spaces.
+ * New lines in the beginning and end are also removed.
+ * Use in tagged templates, e. g.:
+ * ```
+ * documentationBlock`Docs with ${arguments} and more content;`
+ * ```
+ * @param strings Strings in the tagged template. In the example above that would be ['Docs with ', ' and more content;'].
+ * @param args Arguments in the tagged template. In the example above that would be the resolved value for `arguments`;
+ * @returns A string formatted as documentation block.
+ */
+export function documentationBlock(
+  strings: TemplateStringsArray,
+  ...args: string[]
+): string {
+  if (isCodeBlockEmpty(strings.raw, args)) {
+    return '';
+  }
+
+  let adjustedStrings = strings.raw as string[];
+  adjustedStrings = removeSpaceNewLineStartAndEnd(adjustedStrings);
+  adjustedStrings = replaceAllWhiteSpacesBySingleOne(adjustedStrings);
+  adjustedStrings = removeWhiteSpaceAroundNewLine(adjustedStrings);
```

Comment:
The first method is removed and since it zipps the strings are always first and last potentially empty if you start with an empty argument. But I wanted to keep the arguments more or less as they are because they are not used for formatting.